src/objnam.c

```diff
@@ -1000,6 +1000,8 @@ boolean adjective;
 		/* for some reason, this is dragonhide? */
 		if (obj->otyp == LEO_NEMAEUS_HIDE)
 			return "lionhide";
+		if (obj->oartifact && obj->oartifact == ART_XIUHCOATL)
```

Comment:
Asking if `obj->var == const` is equally safe as asking if `obj->var` exists -- both will segfault if `obj` does not exist. That is to say, in general, the check you'd need to use is `if(obj && obj->oartifact == const)`.

In this particular case, we are already assuming that `obj` exists, so no additional checks are necessary, and we can just use `if(obj->oartifact == ART_XIUHCOATL)` directly.